packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts

```diff
@@ -84,15 +110,8 @@ export abstract class S3BucketOrigin extends cloudfront.OriginBase {
         }
 
         if (bucket.encryptionKey) {
-          let bucketName = bucket.bucketName;
-          if (Token.isUnresolved(bucket.bucketName)) {
```

Comment:
Removed this because turns out `bucket.bucketName` is always unresolved as it returns CFN reference to the bucket (i.e. `!Ref bucket`). It is set [here](https://github.com/samson-keung/aws-cdk/blob/4e7d4804093d1a2a54f03830a3770d56c4729bc8/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L1962).